advisories/a82079.md

```diff
@@ -0,0 +1,183 @@
+
+## Statement
+
+This is resolved in CockroachDB by PR [#82087](https://github.com/cockroachdb/cockroach/pull/82087) which fixes the missing removal of back-references to the materialized view.
+
+The fix has been applied to maintenance releases of CockroachDB v21.2.13 and v22.1.1. This public issue is tracked by [#82079](https://github.com/cockroachdb/cockroach/pull/82079).
+
+## Mitigation
+
+Users of CockroachDB versions v21.2.0 to v21.2.12 or v22.1.0 are encouraged to upgrade to [v21.2.13](../releases/v21.2.html#v21-2-13) and [v22.1.1](../releases/v22.1.html#v22-1-1) or a later version. Upgrading to a later maintenance release will prevent this problem from appearing, but it will not fix the problem if it is already present.
+
+If the problem is present according to the diagnostic query above, users can attempt a repair by executing the following:
+
+~~~ sql
+WITH
+	tables
+		AS (
+			SELECT
+				id, tab
+			FROM
+				(
+					SELECT
+						id,
+						crdb_internal.pb_to_json(
+							'cockroach.sql.sqlbase.Descriptor',
+							descriptor
+						)->'table'
+							AS tab
+					FROM
+						system.descriptor
+				)
+			WHERE
+				tab IS NOT NULL
+		),
+	old_depended_on_by_entries
+		AS (
+			SELECT
+				id, json_array_elements(tab->'dependedOnBy') AS entry
+			FROM
+				tables
+		),
+	new_depended_on_by_entries
+		AS (
+			SELECT
+				*
+			FROM
+				old_depended_on_by_entries
+			WHERE
+				(entry->>'id')::INT8 IN (SELECT id FROM system.namespace)
+		),
+	ids_of_descriptors_that_need_upsert
+		AS (
+			SELECT
+				DISTINCT id
+			FROM
+				old_depended_on_by_entries
+			WHERE
+				(entry->>'id')::INT8 NOT IN (SELECT id FROM system.namespace)
+		),
+	new_depended_on_by_arrs
+		AS (
+			(
+				SELECT
+					id,
+					json_agg(entry ORDER BY (entry->>'id')::INT8 ASC)
+						AS new_depended_on_by_arr
+				FROM
+					new_depended_on_by_entries
+				WHERE
+					id IN (SELECT id FROM ids_of_descriptors_that_need_upsert)
+				GROUP BY
+					id
+				UNION
+					SELECT
+						id, '[]'::JSONB AS new_depended_on_by_arr
+					FROM
+						ids_of_descriptors_that_need_upsert
+					WHERE
+						id NOT IN (SELECT id FROM new_depended_on_by_entries)
+			)
+			ORDER BY
+				id
+		)
+SELECT
+	crdb_internal.unsafe_upsert_descriptor(
+		tables.id,
+		crdb_internal.json_to_pb(
+			'cockroach.sql.sqlbase.Descriptor',
+			json_build_object(
+				'table',
+				json_remove_path(
+					json_set(
+						json_set(
+							tab,
+							ARRAY['dependedOnBy'],
+							new_depended_on_by_arr
+						),
+						ARRAY['version'],
+						((tab->>'version')::INT8 + 1)::STRING::JSONB
+					),
+					ARRAY['modificationTime']
+				)
+			)
+		),
+		true
+	)
+FROM
+	tables
+	JOIN new_depended_on_by_arrs ON tables.id = new_depended_on_by_arrs.id;
+~~~
+
+We encourage users to [back up](../{{site.versions["stable"]}}/backup-and-restore-overview.html) their clusters before undertaking such a repair.
```

Comment:
How do I know if my repair attempt succeeded?

Do we have a clearer recommendation? e.g., "if the repair attempt succeeds, X will happen.  If X does not happen, restore from a backup"

basically as a user I would no longer trust my cluster in this state without some more guarantees tbh

Reply:
@postamar this ^^ is really a question for you I think, not really a docs copy edit type comment